frontend/src/components/common/table/table.tsx

```diff
@@ -20,43 +22,51 @@ const Table = <Data extends Record<string, unknown>>({
   const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
     tableInstance;
 
+  const hasData = Boolean(data.length);
+
```

Comment:
You can early return placeholder just to avoid additional nesting
```suggestion
  const hasData = Boolean(data.length);

  if (!hasData) {
    return (
      <p className={styles.placeholder}>{placeholder}</p>
    );
  }

  return <... all table stuff ...>;
```

Reply:
+1 for early return